### PR TITLE
[WIP] add Helm Tiller to Kubernetes addon modules

### LINKED ISSUE #854: Add Helm Tiller as Addon
**Is this a request for help?**:
 
No
---

**Is this an ISSUE or FEATURE REQUEST?** (choose one):
Feature Request

**Orchestrator and version (e.g. Kubernetes, DC/OS, Swarm)**
Kubernetes

**Feature**

Add [http://helm.sh](Helm)'s Tiller (in-cluster) component as an add-on module for Kubernetes (_a la_ Heapster). Tiller runs as a Deployment (with an optional Service) inside of the Kubernetes cluster. The installation should be as close as possible to the `helm init` behavior. Ideally, a user should only have to install the `helm` cli client (`helm init --client-only`), and immediately be able to talk to Tiller (`helm version` should show Tiller's version with no other configuration steps).

Other than image version, there are not many commonly used configuration options on Tiller's deployment, so this should be fairly straightforward.

### PR DESCRIPTION
**Pull request by technosophos**
This implements feature request #854 by adding the Tiller manifest and
supporting configuration to the Kubernetes addon modules.

Closes #854

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

---
This change is [Reviewable](https://reviewable.io/reviews/azure/acs-engine/855)

### CODE REVIEW

**Comment by technosophos** [comment]
Does anyone know why the CLA-bot does not recognize me?

**Comment by anhowe** [comment]
are you signed up with Azure, and ACS?  @colemickens may also have suggestions as to why CLA bot does not recognize you

File: parts/kubernetesmasteraddons-tiller-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        TerminationMessagePolicy: ""
+        VolumeMounts: null
+        WorkingDir: ""
+      DNSPolicy: ""
+      Hostname: ""
+      ImagePullSecrets: null
+      InitContainers: null
+      NodeName: ""
+      NodeSelector:
+        beta.kubernetes.io/os: linux
```
Comment (anhowe) [inline parts/kubernetesmasteraddons-tiller-deployment.yaml]: thanks for adding restriction to linux

File: parts/kubernetesmasteraddons-tiller-deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+      AutomountServiceAccountToken: null
+      Containers:
+      - Args: null
+        Command: null
+        Env:
+        - Name: TILLER_NAMESPACE
+          Value: kube-system
+          ValueFrom: null
+        EnvFrom: null
+        Image: gcr.io/kubernetes-helm/tiller:v2.5.0
```
Comment (anhowe) [inline parts/kubernetesmasteraddons-tiller-deployment.yaml]: replace with `<kubernetesTillerSpec>`

File: pkg/acsengine/const.go [hunk 3]
```diff
@@ -62,6 +62,7 @@ var KubeImages = map[api.OrchestratorVersion]map[string]string{
 		"dnsmasq":      "k8s-dns-dnsmasq-amd64:1.13.0",
 		"pause":        "pause-amd64:3.0",
 		"windowszip":   "v1.6.6intwinnat.zip",
+		"tiller":       "gcr.io/kubernetes-helm/tiller:v2.5.0",
```
Comment (anhowe) [inline pkg/acsengine/const.go]: the full URL is constructed elsewhere, enabling China cloud.  I would just keep this as `tiller:v2.5.0`, and verify it constructs the url correctly

Comment (anhowe) [inline pkg/acsengine/const.go]: repeat this for all of the below

**Comment by technosophos** [comment]
I would not want to add the nginx-ingress on the same PR. It's a separate beast. I'm not totally sure how we will do the config for that one.

File: parts/kubernetesmasteraddons-tiller-deployment.yaml (hunk 1 above)
Comment (technosophos) [inline parts/kubernetesmasteraddons-tiller-deployment.yaml]: That's actually the default for Helm now

**Comment by technosophos** [comment]
Alright, I think I have things fixed up. Still not sure what the CLA-bot is looking for. I checked my email, and I checked to make sure I belong to the Azure GH org... is there something else that it is looking for?

**Comment by anhowe** [comment]
@acs-bot test this please

**Comment by technosophos** [comment]
I don't seem to have access to the Details page for the broken build (I get 404). Can someone paste them here so I can see what I need to fix? Thanks.

**Comment by technosophos** [comment]
Rebased and re-pushed.

**Comment by shrutir25** [comment]
acs-bot test this please

**Comment by seanknox** [comment]
@acs-bot test this please

**Comment by seanknox** [comment]
@acs-bot test this please

**Comment by seanknox** [comment]
Closing in favor of https://github.com/Azure/acs-engine/pull/924 to push this over the finish line.

**Pull request closed without merging**